include/dlaf/factorization/cholesky/mc/cholesky_L.h

```diff
@@ -87,189 +152,88 @@ void cholesky_L(Matrix<T, Device::CPU>& mat_a) {
-        // compute first tile of the column of the trailing matrix: diagonal element mat_a(j,j), reading
-        // mat_a.read(j,k), using herk (blas operation)
-        hpx::dataflow(trailing_matrix_executor, unwrapping(tile::herk<T, Device::CPU>), Lower, NoTrans,
-                      -1.0, panel[i_local], 1.0, mat_a(LocalTileIndex{i_local, j_local}));
-
-        col_panel = panel[i_local];
+    // Iterate over the trailing matrix
+    for (SizeType j = k + 1; j < nrtile; ++j) {
+      GlobalTileIndex jj_idx(j, j);
+      comm::Index2D jj_rank = mat_a.rankGlobalTile(jj_idx);
+
+      // Broadcast the jk-tile along the j-th column and update the jj-tile
+      if (this_rank == jj_rank) {
+        pool_executor trailing_matrix_executor = (j == k + 1) ? executor_hp : executor_normal;
+        herk_trailing_diag_tile(trailing_matrix_executor, panel[j], mat_a(jj_idx));
+        if (j != nrtile - 1)
+          send_tile(executor_mpi, mpi_task_chain, Coord::Col, panel[j]);
       }
-      else {
-        // Avoid useless communications if one-row communicator grid and if on the last panel
-        if (col_comm_size > 1 && j != (mat_a.nrTiles().cols() - 1)) {
-          // Update the (trailing) panel column-wise
-          auto recv_bcast_f =
-              unwrapping([rank = j_rank_row, tile_size = mat_a.tileSize(GlobalTileIndex(j, k))](
-                             auto&& comm_wrapper) -> ConstTile_t {
-                MemView_t mem_view(tile_size.linear_size());
-                Tile_t tile(tile_size, std::move(mem_view), tile_size.rows());
-                comm::sync::broadcast::receive_from(rank, comm_wrapper().colCommunicator(), tile);
-                return std::move(tile);
-              });
-          col_panel = hpx::dataflow(executor_mpi, std::move(recv_bcast_f), serial_comm());
-        }
+      else if (this_rank.col() == jj_rank.col()) {
+        GlobalTileIndex jk_idx(j, k);
+        if (j != nrtile - 1)
+          panel[j] = recv_tile<T>(executor_mpi, mpi_task_chain, Coord::Col, mat_a.tileSize(jk_idx),
+                                  jj_rank.row());
       }
 
-      for (SizeType i_local = distr.nextLocalTileFromGlobalTile<Coord::Row>(j + 1);
-           i_local < localnrtile_rows; ++i_local) {
-        // Update remaining trailing matrix mat_a(i,j), reading mat_a.read(i,k) and mat_a.read(j,k),
-        // using gemm (blas operation)
-        hpx::dataflow(trailing_matrix_executor, unwrapping(tile::gemm<T, Device::CPU>), NoTrans,
-                      ConjTrans, -1.0, panel[i_local], col_panel, 1.0,
-                      std::move(mat_a(LocalTileIndex{i_local, j_local})));
+      for (SizeType i = j + 1; i < nrtile; ++i) {
+        // Update the ij-tile using the ik-tile and jk-tile
+        GlobalTileIndex ij_idx(i, j);
+        comm::Index2D ij_rank = distr.rankGlobalTile(ij_idx);
+        if (this_rank == ij_rank) {
+          gemm_trailing_matrix_tile(executor_normal, panel[i], panel[j], mat_a(ij_idx));
+        }
```

Comment:
Can I suggest to further improve the readability a method called e.g. `herkTrailingMatrix` which performs the two loops (line 212 and line 230) and schedule (full) herk tasks (herks and gemms)?
(This helps reducing code duplication as well, as functions that are introduced can be easily reused in other algorithms).

Similarly we can introduce methods for the panel bcasts (lines 200-228 (BLAS ops excluded)) that can be used in other algorithms (e.g. generalized to standard transformation and the reduction to band need the same communication pattern for trailing matrix her2k)

Reply:
Yes, certainly. I just implemented `herkTrailingMatrix()`, you can have a look.

Reply:
I meant to separate the communication from the computation part, as this communication pattern is found elsewhere (e.g. trailing updates with her2k).
But for now it is ok. Reuse of this pattern is not part of this PR.

Reply:
Did you mean just this part (this is from the local but it's the same for the global)?:

```
179     for (SizeType j = k + 1; j < nrtile; ++j) {
180       // Choose queue priority
181       auto trailing_matrix_executor = (j == k + 1) ? executor_hp : executor_normal;
182
183       // Update trailing matrix: diagonal element mat_a(j,j), reading mat_a.read(j,k), using herk (blas operation)
184       herk_trailing_diag_tile(trailing_matrix_executor, mat_a.read(LocalTileIndex{j, k}),
185                               mat_a(LocalTileIndex{j, j}));
186
187       for (SizeType i = j + 1; i < nrtile; ++i) {
188         // Update remaining trailing matrix mat_a(i,j), reading mat_a.read(i,k) and mat_a.read(j,k),
189         // using gemm (blas operation)
190         gemm_trailing_matrix_tile(trailing_matrix_executor, mat_a.read(LocalTileIndex{i, k}),
191                                   mat_a.read(LocalTileIndex{j, k}), mat_a(LocalTileIndex{i, j}));
192       }
193     }
```

Reply:
Don't worry. This PR generated some ideas in my mind, I will elaborate them to see if they are feasible and discuss them separately.

Reply:
OK, then, should I revert the commit and leave `herkTrailingMatrix()` the function to you?

Reply:
OK.